# RESOLUÇÃO DO PROJETO DE CERTIFICAÇÃO

## Contexto do Problema

O investidor James Bauer gostaria de diversificar seus negócios e começar a investir em imóveis. Ele definiu que compraria imóveis na cidade de Nova York, nos Estados Unidos. Por ser um dos locais mais caros para se viver no País, ele acredita que obterá um retorno satisfatório de seus investimentos caso loque imóveis na cidade. Como todas as suas decisões são tomadas com base em dados, ele contratou você, cientista de dados, para ajudá-lo nessa empreitada.

James Bauer planeja inicialmente locar os imóveis adquiridos e por isso ele definiu que irá utilizar a plataforma Airbnb para esse fim. Para isso, ele lhe entregou uma base de dados públicos da empresa, contendo os dados do comportamento dos hosts e de seus imóveis.

Lembrando que o contexto, pessoas e perguntas são completamente fictícios e existem somente na minha imaginação.

## O Desafio

James Bauer o contratou para realizar o estudo da base de dados fornecida e ajudá-lo com a escolha das regiões onde há maior locação e maiores preços, e que fiquem em regiões favoráveis da cidade de Nova York, pois ele acredita que essas características irão ajudá-lo a recuperar o dinheiro investido na aquisição desses imóveis mais rapidamente. Além disso, James Bauer pretende 

Dessa forma, seu trabalho é realizar uma análise exploratória e responder às seguintes perguntas feitas pelo Sr. James Bauer:

1. Qual o `id` do imóvel com o aluguel (diária) mais caro da base de dados?
2. Qual o `id` do imóvel com o aluguel (diária) mais barato da base de dados?
3. Qual o `id` do imóvel que foi mais locado da base de dados?
4. Qual o `id` do imóvel que ficou mais tempo com o anúncio disponível, em
dias, para locação na base de dados?
5. Qual o `id` do imóvel que ficou menos tempo com o anúncio disponível, em
dias, para locação na base de dados?
6. O imóvel com o maior valor de aluguel (diária) da base de dados é o imóvel
que possui mais avaliações na base de dados?
7. O imóvel que possui a menor quantidade mínima de diárias para locação é
também o imóvel que possui o aluguel mais caro?
8. Qual é a média do número mínimo de diárias para locação de um imóvel? 
9. Qual é o `id` do imóvel com a menor quantidade mínima de diárias para locação da base de dados?
10. Qual é o `id` do host que possui o imóvel mais alugado na base de dados?
11. Qual é o `id` do host que possui o imóvel menos alugado na base de dados?
12. Qual é o `id` do host que possui o imóvel com mais avaliações na base de
dados?
13. Qual é o `id` do host que possui a maior quantidade de imóveis cadastrados
na base de dados?
14. Qual o `id` do host que possui o imóvel com a última avaliação feita na base
de dados?
15. Qual o `id` do host que possui mais imóveis ativos dentro da base de dados?
16. O host que possui mais imóveis é o host que também possui mais
avaliações?
17. Qual é a categoria que mais possui imóveis dentro da base de dados?
18. Qual é a categoria que menos possui imóveis dentro da base de dados?
19. A região de Manhattan é a região que mais possui imóveis ativos para
locação, da categoria `Private room`?
20. Qual a categoria de imóvel que possui a maior média de tempo de
disponibilidade para serem locados da região do `Bronx`? Considere somente
imóveis ativos
21. Qual a categoria de imóvel ativo que possui o maior valor de aluguel (diária)
na região de `Manhattan`?
22. A categoria de imóvel `Private Room` é a categoria que fica mais tempo
disponível para locação?
23. Qual a categoria de imóvel que fica mais tempo disponível, na média, para
locação?
24. A categoria de imóvel que fica menos tempo disponível, na média, para
locação é a categoria que possui, em média, o menor aluguel (diária)?
25. A categoria de imóvel `Entire home/apt` é a categoria que possui, na média, o
maior valor de aluguel?
26. A categoria de imóvel `Entire home/apt` é a categoria que possui, na média,
menos locações?
27. A categoria de imóvel `Private Room` na região de `Manhattan`, na média, é
a categoria que possui o menor valor (diária) de locação, comparado as
outras categorias na mesma região?
28. Qual a região que possui a maior quantidade de imóveis?
29. Qual a região que possui a menor quantidade de imóveis?
30. A região que possui a maior quantidade de imóveis é também a região que
possui os imóveis mais locados?
31. Qual a região que possui a menor quantidade de imóveis locados dentro da
base de dados?
32. A região de `Manhattan` é a região que possui, na média, os maiores
aluguéis (diárias) dentro da base de dados?
33. A região de `Queens` é a região que possui, na média, os menores aluguéis
(diárias) dentro da base de dados?
34. Qual a região que possui, na média, os imóveis com os menores aluguéis
dentro da base de dados?
35. Qual a região que possui os imóveis que ficam, na média, menos tempo
disponíveis para aluguel? Ou seja, são alugados mais rápidos na média?
36. Qual o bairro possui a maior quantidade de imóveis ativos para locação?
37. Qual o bairro possui a menor quantidade de imóveis ativos para locação?
38. Qual bairro está o imóvel com o maior valor de aluguel?
39. Qual bairro está o imóvel com o menor valor de aluguel?
40. Qual região que possui o bairro com mais imóveis ativos disponíveis para
locação?
41. Qual região que possui o bairro com menos imóveis ativos disponíveis para
locação?
42. O Bairro `Upper West Side`, na região de `Manhattan`, é o bairro que possui,
na média, o maior aluguel dentro da base de dados?
43. Qual é o bairro que possui, na média, o aluguel mais caro da base de dados?
44. Qual o id do imóvel que possui a melhor rentabilidade da base de dados?
Levando em consideração o aluguel mais caro e imóvel mais rápido de
alugar? Utilize somente os imóveis que estejam ativos e que possuam pelo
menos uma avaliação. Utilize a seguinte fórmula para verificar qual o melhor
imóvel:

$$ 𝑟𝑒𝑛𝑡𝑎𝑏𝑖𝑙𝑖𝑑𝑎𝑑𝑒 = \frac{𝑝𝑟𝑖𝑐𝑒 * (𝑚𝑖𝑛𝑖𝑚𝑢𝑚\_𝑛𝑖𝑔ℎ𝑡𝑠 + 1) *  n𝑢𝑚𝑏𝑒𝑟\_𝑜𝑓\_𝑟𝑒𝑣𝑖𝑒𝑤𝑠}{ \sqrt{𝑎𝑣𝑎𝑖𝑙𝑎𝑏𝑖𝑙𝑖𝑡𝑦_365}} $$

45. Qual a região em que fica o melhor bairro para se adquirir um imóvel, visando
a melhor rentabilidade média? Utilize como métrica o índice criado na
questão anterior.
46. Qual a região em que fica o pior bairro para se adquirir um imóvel, visando a melhor rentabilidade média? Utilize como métrica o índice criado na questão 44.
47. Levando em conta o bairro encontrado na questão anterior, verifique qual é o
`id` do imóvel ativo que possui a pior rentabilidade desse bairro
48. Levando em conta o bairro da questão 45, verifique qual o `id` do imóvel ativo que possui a melhor rentabilidade desse bairro
49. Caso eu, investidor, compre o imóvel da questão anterior com um
investimento de U$ 1.000.000,00, quantas vezes eu teria que locá-lo para ter
o retorno do investimento feito? Utilize a fórmula abaixo para calcular o tempo
de retorno do investimento:

$$ 𝑟𝑒𝑡𝑢𝑟𝑛_𝑖𝑛𝑣𝑒𝑠𝑡𝑚𝑒𝑛𝑡 = \frac{𝑖𝑛𝑣𝑒𝑠𝑡𝑚𝑒𝑛𝑡}{𝑝𝑟𝑖𝑐𝑒 * (𝑚𝑖𝑛𝑖𝑚𝑢𝑚\_𝑛𝑖𝑔ℎ𝑡𝑠 + 1)} $$

50. Se eu desejasse comprar um imóvel que esteja ativo e com pelo menos uma avaliação em um dos bairros abaixo, qual seria o `id` do imóvel com a melhor rentabilidade dentre esses bairros? Utilize o índice calculado na questão 44.
  * East Harlem
  * Harlem
  * Midtown
  * Morningside Heights
  * Upper West Side
  * Upper East Side

## Os Dados

O conjunto de dados que representam o contexto está disponível na plataforma do Kaggle. O link para acesso aos dados :

https://www.kaggle.com/datasets/dgomonov/new-york-city-airbnb-open-data


## Solução do Desafio

### Imports de Bibliotecas

In [1]:
import pandas as pd
import numpy as np
from IPython.display import Image

pd.options.display.float_format = "{:.2f}".format

### Leitura do Arquivo

In [2]:
df_raw = pd.read_csv(r'C:\Users\andre\repository\python-zero-ao-ds-test\data\AB_NYC_2019.csv')
df_raw.head(5)

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington     40.65     -73.97   
1     Jennifer           Manhattan       Midtown     40.75     -73.98   
2    Elisabeth           Manhattan        Harlem     40.81     -73.94   
3  LisaRoxanne            Brooklyn  Clinton Hill     40.69     -73.96   
4        Laura           Manhattan   East Harlem     40.80     -73.94   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  2018-10-19   
1  Entire home/apt    225               1                 45  2019-05-21   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  2019-07-05   
4  Entire home/apt     80              10                  9  2018-11-19   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

### Entendimento dos dados

In [3]:
# Data diimension
df_raw.shape

(48895, 16)

In [4]:
# Data Types
df_raw.dtypes

id                                  int64
name                               object
host_id                             int64
host_name                          object
neighbourhood_group                object
neighbourhood                      object
latitude                          float64
longitude                         float64
room_type                          object
price                               int64
minimum_nights                      int64
number_of_reviews                   int64
last_review                        object
reviews_per_month                 float64
calculated_host_listings_count      int64
availability_365                    int64
dtype: object

In [5]:
# Missing data
df_raw.isna().sum()

id                                    0
name                                 16
host_id                               0
host_name                            21
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                 10052
calculated_host_listings_count        0
availability_365                      0
dtype: int64

### Tratamento dos Dados

"name" e "host_name" não parecem ser um problema grave. Não precisaremos fazer o tratamento.4

Já as colunas last review e reviewes per month, os nulos aparecem quando não há reviews. Portanto é uma questão do negócio natural, portanto não vamos fazer o tratamento
As for last review and reviews per month, it seems that the null is caused by the number of reviews that is 0, so we will fill the na's with 0.

In [6]:
# Converte Last review para data e 
df_raw['last_review'] = df_raw['last_review'].apply(lambda x: pd.to_datetime(x, format='%Y-%m-%d') if pd.notna(x) 
                                                    else "NaN")

C:\Users\andre\AppData\Local\Temp\ipykernel_17976\2988138662.py:2: FutureWarning: Inferring datetime64[ns] from data containing strings is deprecated and will be removed in a future version. To retain the old behavior explicitly pass Series(data, dtype=datetime64[ns])
  df_raw['last_review'] = df_raw['last_review'].apply(lambda x: pd.to_datetime(x, format='%Y-%m-%d') if pd.notna(x)


### Estatística Descritiva

In [7]:
df_raw.dtypes

id                                         int64
name                                      object
host_id                                    int64
host_name                                 object
neighbourhood_group                       object
neighbourhood                             object
latitude                                 float64
longitude                                float64
room_type                                 object
price                                      int64
minimum_nights                             int64
number_of_reviews                          int64
last_review                       datetime64[ns]
reviews_per_month                        float64
calculated_host_listings_count             int64
availability_365                           int64
dtype: object

In [8]:
# Separando variáveis numéricas e categ[oricas
num_attributes = df_raw.select_dtypes(include=['int64', 'float64'])
cat_attributes = df_raw.select_dtypes(exclude=['int64', 'float64'])

#### Variáveis Numéricas

In [9]:
# Tendências Centrais (media e mediana)
ct1 = pd.DataFrame(num_attributes.apply(np.mean)).T
ct2 = pd.DataFrame(num_attributes.apply(np.median)).T

# Medidas de Dispersão (desvio padrao, minimo, maximo, range, kurtosis, skew)
d1 = pd.DataFrame(num_attributes.min()).T
d2 = pd.DataFrame(num_attributes.max()).T
d3 = pd.DataFrame(num_attributes.max() - num_attributes.min()).T
d4 = pd.DataFrame(num_attributes.apply(np.std)).T
d5 = pd.DataFrame(num_attributes.apply(lambda x: x.skew())).T
d6 = pd.DataFrame(num_attributes.apply(lambda x: x.kurtosis())).T

# Creating DataFrame
statistical_attributes = pd.concat([ct1, ct2, d1, d2, d3, d4, d5, d6]).T.reset_index()

statistical_attributes.columns = ['attribute', 'mean', 'median', 'min', 'max', 'range',
                                  'std_dev', 'skew', 'kurtosis']

statistical_attributes

attribute        mean      median     min  \
0                              id 19017143.24 19677284.00 2539.00   
1                         host_id 67620010.65 30793816.00 2438.00   
2                        latitude       40.73       40.72   40.50   
3                       longitude      -73.95      -73.96  -74.24   
4                           price      152.72      106.00    0.00   
5                  minimum_nights        7.03        3.00    1.00   
6               number_of_reviews       23.27        5.00    0.00   
7               reviews_per_month        1.37         NaN    0.01   
8  calculated_host_listings_count        7.14        1.00    1.00   
9                availability_365      112.78       45.00    0.00   

           max        range     std_dev  skew  kurtosis  
0  36487245.00  36484706.00 10982996.07 -0.09     -1.23  
1 274321313.00 274318875.00 78610163.15  1.21      0.17  
2        40.91         0.41        0.05  0.24      0.15  
3       -73.71         0.53        0.05  1.28      5.02  
4     10000.00     10000.00      240.15 19.12    585.67  
5      1250.00      1249.00       20.51 21.83    854.07  
6       629.00       629.00       44.55  3.69     19.53  
7        58.50        58.49        1.68  3.13     42.49  
8       327.00       326.00       32.95  7.93     67.55  
9       365.00       365.00      131.62  0.76     -1.00

### Feature Engineering

In [10]:
df1 = df_raw.copy()

In [11]:
# Disponibilidade
df1['availability'] = df1['availability_365'].apply(lambda x: 1 if x > 1 else 0)

# Days since last review
max_last_review = df1['last_review'].max()
df1['days_since_last_review'] = (max_last_review - df1['last_review']) / np.timedelta64(1, 'D')

In [12]:
df1.sample(3)

id                                               name   host_id  \
10084   7733896  Big Master bedroom with attached Private Bathroom   1036161   
43362  33611988   Spacious,Quite,beautiful room in Astoria-Ditmars  64453547   
40525  31413351  Private entrance, own bathroom, comfy  & sunshine  33614329   

      host_name neighbourhood_group       neighbourhood  latitude  longitude  \
10084     Blake            Brooklyn  Bedford-Stuyvesant     40.68     -73.95   
43362      Onur              Queens    Ditmars Steinway     40.78     -73.91   
40525    Walter            Brooklyn            Bushwick     40.70     -73.93   

             room_type  price  minimum_nights  number_of_reviews last_review  \
10084     Private room    220               1                 88  2019-06-02   
43362     Private room     53              20                  0         NaT   
40525  Entire home/apt     85               1                 36  2019-07-03   

       reviews_per_month  calculated_host_listings_count  availability_365  \
10084               1.84                               2                60   
43362                NaN                               3                 0   
40525               6.10                               4               210   

       availability  days_since_last_review  
10084             1                   36.00  
43362             0                     NaN  
40525             1                    5.00

#### Respondendo as Perguntas do Investidor

##### 1. Qual o `id` do imóvel com o aluguel (diária) mais caro da base de dados

In [13]:
# Seleciono o maior preço
max_price = df1['price'].max()

# Seleciono os imóveis com maior preço. Como temos mais de um, uso como critério a disponibilidade
r1 = df1.loc[(df1['price'] == max_price)].sort_values(by='id', ascending=True).reset_index(drop=True).loc[0, 'id']
r1

7003697

**id: 7003697**

##### 2. Qual o `id` do imóvel com o aluguel (diária) mais barato da base de dados? Excluo aqueles que estão com valor de diária = 0

In [14]:
# Seleciono o menor preço
min_price = df1[df1['price'] != 0]['price'].min()

# Seleciono os imóveis com menor preço. Como temos mais de um, aplicamos o critério da disponibilidade
r2 = df1.loc[(df1['price'] == min_price)].sort_values(by='id', ascending=True).reset_index(drop=True).loc[0, 'id']
r2

1620248

**id: 1620248**

##### 3. Qual o `id`  do imóvel que foi mais locado da base de dados?*

In [15]:
r3 = df1[['id', 'number_of_reviews']].sort_values('number_of_reviews', ascending=False).reset_index(drop=True).loc[0, 'id']
r3

9145202

**id: 9145202**

##### 4. Qual o `id`  do imóvel que ficou mais tempo com o anúncio disponível, em dias, para locação na base de dados?*

In [16]:
r4 = df1[['id', 'days_since_last_review']].sort_values('days_since_last_review', 
                                                        ascending=False).reset_index(drop=True).loc[0, 'id']
r4

74860

**id: 74860** 

##### 5. Qual o `id` do imóvel que ficou menos tempo com o anúncio disponível, em dias, para locação na base de dados?

In [17]:
r5 = df1[['id', 'days_since_last_review']].sort_values(['days_since_last_review', 'id'], 
                                                       ascending=True).reset_index(drop=True).loc[0, 'id']
r5

100184

**id: 100184**

##### 6. O imóvel com o maior valor de aluguel (diária) da base de dados é o imóvel que possui mais avaliações na base de dados?*

In [18]:
r6 = df1[['id', 'number_of_reviews']].sort_values(by='number_of_reviews', ascending=False).reset_index(drop=True).loc[0, 'id']

if r6 == r1:
    print('Verdadeiro')
else: 
    print('Falso')

Falso


**Falso** 

##### 7. O imóvel que possui a menor quantidade mínima de diárias para locação é também o imóvel que possuí o aluguel mais caro?

In [19]:
min_minimum_nights = df1['minimum_nights'].min()

r7 = df1[df1['minimum_nights'] == min_minimum_nights].sort_values(by='id', ascending=True).reset_index(drop=True).loc[0, 'id']
r7

if r7 == r1:
    print('Verdadeiro')
else:
    print('Falso')

Falso


**Falso**

##### 8. Qual é a média do número mínimo de diárias para locação de um imóvel? 

In [20]:
r8 = round(np.mean(df1['minimum_nights']), 2)
r8

7.03

**7.03**

##### 9. Qual é o `id` do imóvel com a menor quantidade mínima de diárias para locação da base de dados?

In [21]:
r9 = r7
r9 

2539

**id: 2539**

##### 10. Qual é o `id` do host que possui o imóvel mais alugado na base de dados?

In [22]:
r10 = df1.loc[df1['id'] == r3,'host_id'].values[0]
r10

47621202

**host_id: 47621202**

##### 11. Qual é o `id` do host que possui o imóvel menos alugado na base de dados?

In [23]:
r11 = df1[df1['number_of_reviews'] == 1].sort_values(by='host_id', 
                                                     ascending=True).reset_index(drop=True).loc[0, 'host_id']
r11

2438

**host_id: 2438**

##### 12. Qual é o `id` do host que possui o imóvel com mais avaliações na base de dados?

In [24]:
r12 = r10
r12

47621202

**R: Host 47621202**

##### 13. Qual é o `id` do host que possui a maior quantidade de imóveis cadastrados na base de dados?

In [25]:
r13 = df1[['id', 'host_id']].groupby('host_id').count().sort_values(by='id', 
                                                                    ascending=False).reset_index(drop=False).loc[0, 'host_id']
r13

219517861

**R: Host: 219517861** 

##### 14. Qual o `id` do host que possui o imóvel com a última avaliação feita na base de dados?

In [26]:
r14 = df1[df1['last_review'] == max_last_review].sort_values(by='host_id').reset_index(drop=True).loc[0, 'host_id']
r14

149047

**R: host_id: 149047**

##### 15. Qual o `id` do host que possui mais imóveis ativos dentro da base de dados?

In [27]:
r15 = df1[['host_id', 'availability']].groupby('host_id').count().sort_values(by='availability',
                                                                              ascending=False).reset_index().loc[0, 'host_id']
r15

219517861

**host_id: 219517861**

##### 16. O host que possui mais imóveis é o host que também possui mais avaliações?

In [28]:
if r12 == r13:
    r16 = 'Verdadeiro'
else:
    r16 = 'Falso'

print(r16)

Falso


**Falso**

##### 17. Qual é a categoria que mais possui imóveis dentro da base de dados?

In [29]:
r17 = df1[df1['availability'] == 1]
r17 = r17[['id', 'room_type']].groupby('room_type').count().sort_values(by='id', 
                                                                        ascending=False).reset_index().loc[0, 'room_type']
r17

'Entire home/apt'

**Entire home/apt**

##### 18. Qual é a categoria que menos possui imóveis dentro da base de dados?

In [30]:
r18 = df1[df1['availability'] == 1]
r18 = r18[['id', 'room_type']].groupby('room_type').count().sort_values(by='id').reset_index().loc[0, 'room_type']
r18

'Shared room'

**Shared room**

##### 19. A região de `Manhattan` é a região que mais possui imóveis ativos para locação, da categoria `Private room`?

In [31]:
r19 = df1[(df1['room_type'] == 'Private room') & (df1['availability'] == 1)]

resposta = r19[['id', 'neighbourhood_group']].groupby('neighbourhood_group').count().sort_values(by='id', ascending=False).reset_index().loc[0, 'neighbourhood_group']

if resposta == 'Manhattan':
    r19 = 'Verdadeiro'
else:
    r19 = 'Falso'
    
r19

'Falso'

**Falso**

##### 20. Qual a categoria de imóvel que possui a maior média de tempo de disponibilidade para serem locados da região do `Bronx`? Considere somente imóveis ativos

In [32]:
r20 = df1[(df1['neighbourhood_group'] == 'Bronx') & (df1['availability'] == 1)]
r20 = r20[['room_type', 'days_since_last_review']].groupby('room_type').mean().sort_values(by='days_since_last_review', ascending=False)
r20 = r20.reset_index().loc[0, 'room_type']
r20

'Shared room'

**Shared room**

##### 21. Qual a categoria de imóvel ativo que possui o maior valor de aluguel (diária) na região de `Manhattan`?

In [33]:
r21 = df1[(df1['neighbourhood_group'] == 'Manhattan') & (df1['availability'] == 1)]
r21[['price', 'room_type']].groupby('room_type').mean().sort_values(by='price', ascending=False)

price
room_type             
Entire home/apt 269.59
Private room    128.44
Shared room      84.72

##### 22. A categoria de imóvel “Private Room“ é a categoria que fica mais tempo disponível para locação?

In [34]:
resposta = df1[['room_type', 'days_since_last_review']].groupby('room_type').mean().sort_values(by='days_since_last_review', ascending=False)
resposta = resposta.reset_index().loc[0, 'room_type']

if resposta == 'Private room':
    r22 = 'Verdadeiro'
else:
    r22 = 'Falso'

r22

'Falso'

**Falso**

##### 23. Qual a categoria de imóvel que fica mais tempo disponível, na média, para locação?

In [35]:
r23 = df1[['room_type', 'days_since_last_review']].groupby('room_type').mean().sort_values(by='days_since_last_review', ascending=False)
r23 = r23.reset_index().loc[0, 'room_type']
r23

'Entire home/apt'

**Entire home/apt**

##### 24. A categoria de imóvel que fica menos tempo disponível, na média, para locação é a categoria que possuí, em média, o menor aluguel (diária)?

In [36]:
menos_tempo_disp = df1[['room_type', 'days_since_last_review']].groupby('room_type').mean().sort_values(by='days_since_last_review').reset_index().loc[0, 'room_type']

menor_aluguel = df1[['price', 'room_type']].groupby('room_type').mean().sort_values(by='price').reset_index().loc[0, 'room_type']

if menos_tempo_disp == menor_aluguel:
    r24 = 'Verdadeiro'
else:
    r24 = 'Falso'

r24

'Verdadeiro'

**Verdadeiro**

##### 25. A categoria de imóvel `Entire home/apt` é a categoria que possui, na média, o maior valor de aluguel?

In [37]:
resposta = df1[['price', 'room_type']].groupby('room_type').mean().sort_values(by='room_type', ascending=True).reset_index().loc[0, 'room_type']

if resposta == 'Entire home/apt':
    r25 = 'Verdadeiro'
else:
    r25 = 'Falso'
    
r25

'Verdadeiro'

**Verdadeiro**

##### 26. A categoria de imóvel `Entire home/apt` é a categoria que possui, na média, menos locações?

In [38]:
resposta = df1[['room_type', 'number_of_reviews']].groupby('room_type').mean().sort_values(by='room_type', ascending=False).reset_index().loc[0, 'room_type']

if resposta == 'Entire home/apt':
    r26 = 'Verdadeiro'
else:
    r26 = 'Falso'
    
r26

'Falso'

**Verdadeiro**

##### 27. A categoria de imóvel `Private Room` na região de Manhattan, na média, é a categoria que possui o menor valor de locação, comparado as outras categorias na mesma região? 

In [39]:
resposta = df1[df1['neighbourhood_group'] == 'Manhattan']

resposta = resposta[['price', 'room_type']].groupby('room_type').mean().sort_values(by='price').reset_index().loc[0, 'room_type']

if resposta == 'Private room':
    r27 = 'Verdadeiro'
else:
    r27 = 'Falso'
    
r27

'Falso'

**Falso**

##### 28. Qual a região que possui a maior quantidade de imóveis?

In [40]:
r28 = df1[['id', 'neighbourhood_group']].groupby('neighbourhood_group').count().sort_values(by='id', ascending=False).reset_index().loc[0, 'neighbourhood_group']
r28

'Manhattan'

**Manhattan**

##### 29. Qual a região que possui a menor quantidade de imóveis?

In [41]:
r28 = df1[['id', 'neighbourhood_group']].groupby('neighbourhood_group').count().sort_values(by='id').reset_index().loc[0, 'neighbourhood_group']
r28

'Staten Island'

**Staten Island**

##### 30. A região que possui a maior quantidade de imóveis é também a região que possui os imóveis mais locados?

In [42]:
mais_imoveis = r28

mais_locados = df1[['neighbourhood_group', 'number_of_reviews']].groupby('neighbourhood_group').sum().sort_values(by='number_of_reviews', ascending=False).reset_index().loc[0, 'neighbourhood_group']

if mais_imoveis == mais_locados:
    r30 = 'Verdadeiro'
else:
    r30 = 'Falso'
    
r30

'Falso'

**Falso**

##### 31. Qual a região que possui a menor quantidade de imóveis locados dentro da base de dados?

In [43]:
r31 = df1[['neighbourhood_group', 'number_of_reviews']].groupby('neighbourhood_group').sum().sort_values(by='number_of_reviews').reset_index().loc[0, 'neighbourhood_group']
r31

'Staten Island'

**Staten Island**

##### 32. A região de `Manhattan` é a região que possui, na média, os maiores aluguéis (diárias) dentro da base de dados?

In [44]:
resposta = df1[['neighbourhood_group', 'price']].groupby('neighbourhood_group').mean().sort_values(by='price', ascending=False).reset_index().loc[0, 'neighbourhood_group']

if resposta == 'Manhattan':
    r32 = 'Verdadeiro'
else:
    r32 = 'Falso'
    
r32

'Verdadeiro'

**Verdadeiro**

##### 33. A região de `Queens` é a região que possui, na média, os menores aluguéis (diárias) dentro da base de dados?

In [45]:
resposta = df1[['neighbourhood_group', 'price']].groupby('neighbourhood_group').mean().sort_values(by='price').reset_index().loc[0, 'neighbourhood_group']

if resposta == 'Queens':
    r33 = 'Verdadeiro'
else:
    r33 = 'Falso'
    
r33

'Falso'

**Falso**

##### 34. Qual a região que possui, na média, os imóveis com os menores aluguéis dentro da base de dados?

In [46]:
r34 = df1[['neighbourhood_group', 'price']].groupby('neighbourhood_group').mean().sort_values(by='price').reset_index().loc[0, 'neighbourhood_group']
r34

'Bronx'

**Bronx**

##### 35. Qual a região que possui os imóveis que ficam, na média, menos tempo disponíveis para aluguel? Ou seja, são alugados mais rápidos na média?

In [52]:
r35 = df1[['neighbourhood_group', 'days_since_last_review']].groupby('neighbourhood_group').mean().sort_values(by='days_since_last_review').reset_index().loc[0, 'neighbourhood_group']
r35

'Staten Island'

**Staten Island**

##### 36. Qual o bairro possui a maior quantidade de imóveis ativos para locação?

In [61]:
r36 = df1[df1['availability'] == 1]
r36 = r36[['id', 'neighbourhood']].groupby('neighbourhood').count().sort_values(by='id', ascending=False).reset_index().loc[0, 'neighbourhood']
r36

'Bedford-Stuyvesant'

**Bedford-Stuyvesant**

##### 37. Qual o bairro possui a menor quantidade de imóveis ativos para locação?

In [63]:
r37 = df1[df1['availability'] == 1]
r37 = r37[['id', 'neighbourhood']].groupby('neighbourhood').count().sort_values(by='id').reset_index().loc[0, 'neighbourhood']
r37

'Silver Lake'

**Silver Lake**

##### 38. Qual bairro está o imóvel com o maior valor de aluguel?

In [70]:
r38 = df1[df1['id'] == r1].loc[:, 'neighbourhood'].values[0]
r38

'Astoria'

**Astoria**

##### 39. Qual bairro está o imóvel com o menor valor de aluguel?

In [72]:
r39 = df1[df1['id'] == r2].loc[:, 'neighbourhood'].values[0]
r39

'East Village'

**East Village**

##### 40. Qual região que possui o bairro com mais imóveis ativos disponíveis para locação?

In [91]:
r40 = df1[df1['availability'] == 1]
r40 = pd.DataFrame(r40[['id', 'neighbourhood_group', 'neighbourhood']].groupby(['neighbourhood', 'neighbourhood_group']).count().sort_values(by='id', ascending=False).reset_index().loc[0, :]).T
r40 = r40.loc[0, 'neighbourhood_group']
r40

'Brooklyn'

**Brooklyn**

##### 41. Qual região que possui o bairro com menos imóveis ativos disponíveis para locação?

In [94]:
r41 = df1[df1['availability'] == 1]
r41 = pd.DataFrame(r41[['id', 'neighbourhood_group', 'neighbourhood']].groupby(['neighbourhood', 'neighbourhood_group']).count().sort_values(by='id').reset_index().loc[0, :]).T
r41 = r41.loc[0, 'neighbourhood_group']
r41

'Staten Island'

**Staten Island**

##### 42. O Bairro `Upper West Side`, na região de `Manhattan`, é o bairro que possui, na média, o maior aluguel dentro da base de dados?

##### 43. Qual é o bairro que possui, na média, o aluguel mais caro da base de dados?

##### 44. Qual o `id` do imóvel que possui a melhor rentabilidade da base de dados? Levando em consideração o aluguel mais caro e imóvel mais rápido de alugar?  Utilize somente os imóveis que estejam ativos e que possuam pelo menos uma avaliação.

Utilize a seguinte fórmula para verificar qual o melhor imóvel:

$$ rentability = \frac{price * (minimoum\_nights + 1) * number\_of\_reviews}{\sqrt{availability\_365}} $$

##### 45. Qual a região em que fica o melhor bairro para se adquirir um imóvel, visando a melhor rentabilidade média? Utilize como métrica o índice criado na questão anterior.

##### 46. Qual a região em que fica o pior bairro para se adquirir um imóvel, visando a melhor rentabilidade média? Utilize como métrica o índice criado na questão 44.

##### 47. Levando em conta o bairro encontrado na questão anterior, verifique qual é o `id` do imóvel ativo que possui a pior rentabilidade desse bairro

##### 48. Levando em conta o bairro da questão 45, verifique qual o `id` do imóvel ativo que possui a melhor rentabilidade desse bairro

##### 49. Caso eu, investidor, compre o imóvel da questão anterior com um investimento de U\$ 1.000.000,00, quantas vezes eu teria que locá-lo para ter o retorno do investimento feito? Utilize a fórmula abaixo para calcular o tempo de retorno do investimento

$$ return\_investment = \frac{investment}{price * (minimum\_nights + 1)} $$

##### 50. Se eu desejasse comprar um imóvel que esteja ativo e com pelo menos uma avaliação em um dos bairros abaixo, qual seria o `id` do imóvel com a melhor rentabilidade dentre esses bairros? Utilize o índice calculado na questão 44.

* East Harlem
* Harlem
* Midtown
* Morningside Heights
* Upper West Side
* Upper East Side

## Próximos Passos

Como próximos passos para deixar o seu projeto um projeto de portifólio, crie um mapa, utilizando as técnicas aprendidas dentro do curso de Python do Zero ao DS, para demonstrar a densidade de imóveis por região ou bairro. 

Coloque também esse projeto em produção com o mapa desenhado e com quaisquer outras alterações ou melhorias que você queira! Não se esqueça de colocar o seu projeto no seu portifólio quando você terminar de melhorá-lo!